v2.0/src/Components/Pages/EventsPage/EventsPageReal.js

```diff
@@ -178,23 +183,73 @@ class EventsPage extends React.Component {
     }
 
     if (events) {
-      return events.map((event) => {
+      const page = events.map((event) => {
+        let recurringDetailString = "";
         const dateString = dateFormatString(
           new Date(event.start_date_and_time),
           new Date(event.end_date_and_time)
         );
-
+        if (event.is_recurring) {
```

Comment:
Scary scary stuff.
Hi @isha-san , I know you have good intentions ,  but are we sure this is the only way to implement the recurring events...? Making api calls in the middle of rendering?